### PR TITLE
Fixvecmul

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
The amat in the MulVec signature is temporary. Again, trying to keep changes minimal. The next commit will return that to a and fix some of the code organization.

**Comment by btracey** [comment]
Wow. The github diffs on this are terrible. The changes are way more simple than they look.

**Comment by kortschak** [comment]
Github diffs don't ignore whitespace. Yes, they are horrible.

**Comment by btracey** [comment]
Ping

File: mat64/list_test.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ func makeRandOf(a Matrix, m, n int) Matrix {
 			return &Vector{
 				n: 0,
 				mat: blas64.Vector{
-					Inc: 1,
+					Inc: 0,
```
Comment (kortschak) [inline mat64/list_test.go]: Just return &Vector{}; everything is zero, so there's no reason to state the zeros.

File: mat64/pool.go [hunk 2]
```diff
@@ -78,3 +78,17 @@ func getWorkspace(r, c int, clear bool) *Dense {
 func putWorkspace(w *Dense) {
 	pool[bits(uint64(cap(w.mat.Data)))].Put(w)
 }
+
+// getWorkspaceVec returns a *Vec of length n and a cap that is less than 2*n. If clear is true, the
+// data slice visible through the Matrix interface is zeroed.
+func getWorkspaceVec(n int, clear bool) *Vector {
+	// TODO(btracey): Replace this when there is a resolution to issue 160.
```
Comment (kortschak) [inline mat64/pool.go]: The resolution is to do the change, but in another PR. Same below.

File: mat64/vector.go [hunk 3]
```diff
@@ -188,48 +188,30 @@ func (v *Vector) DivElemVec(a, b *Vector) {
...
-
-		w.mat.Inc = 1
-		w.n = ar
-		if trans {
-			w.n = ac
-		}
+	a, trans := untranspose(aMat)
+	ar, ac := a.Dims()
+	v.reuseAs(r)
+	var w *Vector
```
Comment (kortschak) [inline mat64/vector.go]: Will this go away and we use v as you have done in the Dense equivalents? Presumably in the same PR that changes aMat.

**Comment by kortschak** [comment]
LGTM with minor changes.

File: mat64/vector.go (hunk 3 above)
Comment (btracey) [inline mat64/vector.go]: Yes, that's the plan.

File: mat64/list_test.go (hunk 1 above)
Comment (btracey) [inline mat64/list_test.go]: Done.

File: mat64/pool.go (hunk 2 above)
Comment (btracey) [inline mat64/pool.go]: Agreed.

**Comment by btracey** [comment]
Only fails because of beta3 bug.

**Pull request merged**